libavcodec/vaapi_decode.h

```diff
@@ -45,6 +45,10 @@ typedef struct VAAPIDecodePicture {
     int                nb_slices;
     VABufferID           *slice_buffers;
     int                   slices_allocated;
+
+    VASurfaceID           sfc_surface;
+    VARectangle           src;
```

Comment:
why we need this, can we put it as a local variable?

Reply:
It's address will be set into vaCreateBuffer, and the api will copy it into a vaBuffer, so we need to keep the var before the current frame decoded.